A55-xds-stateful-session-affinity.md

```diff
@@ -0,0 +1,480 @@
+that works with Envoy will continue to work with proxyless gRPC without any
+modifications.
+
+The design involves following 5 parts.
+
+### `XdsClient` to Include Endpoint Health Status
+
+The `EdsUpdate` struct from the `XdsClient` will be enhanced to include the
+[endpoint health status][eds-health-status] for each endpoint. `XdsClient`
+will also be modified to include endpoints in `DRAINING` state in the
+`EdsUpdate` reported to the watchers.
+
+### Listener and Route Configuration through xDS
+
+The feature is enabled and configured using the
+[cookie based stateful session extension][cookie-ext] which
+is an [http-filter][] in the `HttpConnectionManager` for the target service.
+A sample configuration is as follows:
+
+```textproto
+{ // HttpConnectionManager
+  http_filters: [
+    {
+      name: "envoy.filters.http.stateful_session"
+      typed_config: {
+        "@type": "type.googleapis.com/envoy.extensions.filters.http.stateful_session.v3.StatefulSession"
+        session_state: {
+          name: "envoy.http.stateful_session.cookie"
+          typed_config: {
+            "@type": "type.googleapis.com/envoy.extensions.http.stateful_session.cookie.v3.CookieBasedSessionState"
+            cookie: {
+              name: "global-session-cookie"
+              path: "/Package1.Service2/Method3"
+              ttl: "120s"
+            }
+          }
+        }
+      }
+    }
+  ]
+}
+```
+
+`name` is the name of the cookie used in this feature.
+
+`path` is the path for which this cookie is applied. **Note:** Stateful
+session affinity will not work if the cookie's path does not match the
+request's path. For example, if a given route matches requests with path
+`/service/foo` but the corresponding per-route filter config specifies a
+cookie path of `/service/bar`, then the cookie will never be set.
+
+`ttl` is the time-to-live for the cookie. It is set on the cookie but will
+not be enforced by gRPC.
+
+This new filter will be added to the HTTP Filter Registry and processed as
+described in [A39: xDS HTTP Filter Support][A39]. Specifically when this filter
+is present in the configuration, gRPC will create the appropriate client filter
+(aka interceptor in Java/Go) and install it in the channel to process data
+plane RPCs. We call this filter `CookieBasedStatefulSessionFilter`. It will
+copy the 3 configuration values - `name`, `ttl`  and `path` -  into the filter
+object. Validation and processing are further described in
+[Stateful session][stateful-session].
+
+Note that
+[StatefulSessionPerRoute][stateful-session-per-route] *will* be supported as
+[Filter Config Overrides][filter_config_override]. For example, let's say
+the top level config (in `HttpConnectionManager.http_filters`) contains certain
+`StatefulSession` configuration. This configuration can either be
+overridden or disabled by a
+[StatefulSessionPerRoute][stateful-session-per-route] config for a particular
```

Comment:
'may be supported' (depending on control plane ) ?

Reply:
This document is talking about what the client will support, not what the control plane supports.